notification/events.go

```diff
@@ -84,43 +84,36 @@ type notificationHandler struct {
 	Ring *events.EventRing
 }
 
-// Check if notification can be send in the interval based on group by, returns true if it can sent
+// Check if notification can be sent in the interval based on group by, returns true if it can be sent
 func checkRepeatInterval(ctx context.Context, n NotificationWithSpec, event models.Event) (bool, error) {
 	validKeys := map[string]string{
-		"notification_id": n.ID.String(),
-		"resource_id":     event.Properties["id"],
-		"source_event":    event.Name,
+		"resource_id":  event.Properties["id"],
+		"source_event": event.Name,
 	}
-	var clauses []clause.Expression
+
+	clauses := []clause.Expression{clause.Eq{Column: "notification_id", Value: n.ID.String()}}
 	for _, g := range n.GroupBy {
 		if val, exists := validKeys[g]; exists {
 			clauses = append(clauses, clause.Eq{Column: g, Value: val})
+		} else {
```

Comment:
I think we should just group by resource-id, event and notification id by default and handling custom grouping later